integration/spark-common/src/main/scala/org/apache/carbondata/events/LoadEvents.scala

```diff
@@ -42,6 +42,14 @@ case class LoadTablePostExecutionEvent(sparkSession: SparkSession,
     carbonTableIdentifier: CarbonTableIdentifier,
     carbonLoadModel: CarbonLoadModel) extends Event with LoadEventInfo
 
+/**
+ * Class for handling operations after data load completion and before final commit of load
+ * operation. Example usage: For loading pre-aggregate tables
+ */
+case class LoadTablePreStatusUpdateEvent(sparkSession: SparkSession,
```

Comment:
The usage of this is not clear, why is it needed

Reply:
This event will be used to perform some task just before updating the carbontable status.
For example: When loading data into parent table we need to start load for all child datamaps  so that if any of the load fails then parent table status file would not be written.